diff --git a/lpck_structure.api.inc b/lpck_structure.api.inc
index 99ca0c6..6b8bf4d 100755
--- a/lpck_structure.api.inc
+++ b/lpck_structure.api.inc
@@ -86,13 +86,10 @@ function lpck_structure_get_structure_configs($conditions = array(), $fields = a
  *         Array of LPCK structure objects fetched
  */
 function lpck_structure_lpck_type_get_structure_config($lpt_name) {
-  $conditions = array(
-    array(
-      'name' => 'lpt_name',
-      'value' => $lpt_name,
-    ),
-  );
-  $result = lpck_structure_get_structure_configs($conditions);
+  if(!function_exists('ctools_export_load_object')){
+    ctools_include('export');
+  }
+  $result = ctools_export_load_object('lpck_structure_conf', 'conditions', array('lpt_name' => $lpt_name));
   foreach ($result as $key => $lpsc) {
     if ($lpsc->lpt_name == $lpt_name) {
       $lpsc->settings = unserialize($lpsc->settings);
@@ -153,8 +150,8 @@ function lpck_structure_save_structure(&$lps) {
  */
 function lpck_structure_delete_structure(&$lp) {
   return db_delete('lpck_structure')
-      ->condition('lp_name', $lp)
-      ->execute();
+    ->condition('lp_name', $lp)
+    ->execute();
 }
 
 /**
